### ISSUE TITLE
How can I load decoder from a Machine Translation model to my Speech to text translation model?

### ISSUE THREAD

**Issue by amirfaghihi** [pytorch/fairseq#3769]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I'm training a speech to text model using COVOST dataset, just like it is written in the examples. There is a command tool parameter (**--load-pretrained-encoder-from**) which initializes the encoder of my speech-to-text model with a pre-trained ASR model. I was wondering if there's a way to do the same with the decoder of  ST network and a pretrained Machine translation model. 

I know that to be able to load the 'encoder' or 'decoder' of another pretrained model, the architectures should be the same. For ASR there's already a pretrained model using **"st_transformer_s"** architecture, which matches with the ST model. Another complication is that MT architectures don't match. 

I would appreciate any kind of help. Thanks. 

#### Code

fairseq-train ${COVOST_ROOT}/fr \
  --config-yaml config_st_fr_en.yaml --train-subset train_st_fr_en --valid-subset dev_st_fr_en \
  --save-dir ${ST_SAVE_DIR} --num-workers 4 --max-update 30000 --max-tokens 40000 \  # --max-tokens 50000 for en-*
  --task speech_to_text --criterion label_smoothed_cross_entropy --label-smoothing 0.1 --report-accuracy \
  --arch s2t_transformer_s --encoder-freezing-updates 1000 --optimizer adam --lr 2e-3 \
  --lr-scheduler inverse_sqrt --warmup-updates 10000 --clip-norm 10.0 --seed 1 --update-freq 8 \
  _--load-pretrained-encoder-from ${ASR_SAVE_DIR}/${CHECKPOINT_FILENAME}_

#### What have you tried?

I have searched all over the code to find an option that lets me load pretrained decoder to my model. Seems like there IS such an option from other tasks, just not for ST.

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 1.0.0a0+fc391ff
 - PyTorch Version (e.g., 1.0): 1.4.0
 - OS (e.g., Linux): Linux
 - How you installed fairseq (`pip`, source): source
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:
